skbio/io/util.py

```diff
@@ -42,7 +47,17 @@ def _get_filehandle(filepath_or, *args, **kwargs):
     pass through.
     """
     if _is_string_or_bytes(filepath_or):
-        fh, own_fh = open(filepath_or, *args, **kwargs), True
+        if requests.compat.urlparse(filepath_or).scheme in {'http', 'https'}:
+            sess = CacheControl(requests.Session(),
+                                cache=FileCache(gettempdir()))
+            req = sess.get(filepath_or, **kwargs)
+
+            # if the response is not 200, an exception will be raised
+            req.raise_for_status()
+
+            fh, own_fh = BytesIO(req.content), True
```

Comment:
Turns out I was wrong, even if you pass `stream=True`, all the data will be read into memory [because we were accessing the `.contents` property](http://docs.python-requests.org/en/latest/user/advanced/#blocking-or-non-blocking), the only way we can retrieve the data chunk by chunk is if we [use the `iter_lines` method](http://docs.python-requests.org/en/latest/user/advanced/#streaming-requests). The downside to this is that we cannot (I don't know of a way to do this) put this iterator into a file-like object, unless we roll our own wrapper.

The other thing I tried was to (1) generate the request, (2) read all the contents into `_`, (3) let CacheControl write the data into disk and (4) return a regular file handle to the cached file. Unfortunately this doesn't seem to work very well, I'm still trying to figure out why this happens, I've opened an issue on their repo https://github.com/ionrock/cachecontrol/issues/77